src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -231,9 +231,15 @@ case class Year(digits: Int, nMissingDigits: Int = 0) extends Interval {
   * As with Year, the optional second parameter allows YearSuffix to represent decades (nMissingDigits=1),
   * centuries (nMissingDigits=2), etc.
   */
-case class YearSuffix(interval: Interval, lastDigits: Int, nMissingDigits: Int = 0) extends Interval {
+case class YearSuffix(interval: Interval, lastDigits: Int, nlastDigits: Int = 0, nMissingDigits: Int = 0) extends Interval {
```

Comment:
I gather the problem here is that the calculation `nSuffixDigits` is doing to infer the number of digits that needs to be replaced is wrong with leading zeros. But why not just make `nSuffixDigits` a (non-optional) parameter and force everyone to supply that? It seems like that's what `nlastDigits` is trying to do in a roundabout way.

Reply:
Ok, much simpler. I was trying to keep the calculation of nSuffixDigits just in case.